consensus/src/dag/dag_store.rs

```diff
@@ -215,19 +227,22 @@ impl Dag {
 
     pub fn reachable(
         &self,
-        from: &Arc<CertifiedNode>,
+        targets: &[NodeMetadata],
         until: Option<Round>,
+        filter: impl Fn(&NodeStatus) -> bool,
```

Comment:
instead of passing a pattern match I think we can simply pass a bool indicating whether to only include ordered or not

Reply:
but this gives more flexibility? One can also filter based on other node fields as well -- not saying its needed but good to have? I will add a TODO to clean it up if we end up not using it for anything else.